ui/components/ui/icon-button/icon-button.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { forwardRef } from 'react';
+import classNames from 'classnames';
+import { Text, ButtonBase } from '../../component-library';
+import type { ButtonBaseProps } from '../../component-library/button-base/button-base.types';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import Tooltip from '../tooltip/tooltip';
+
+export type IconButtonProps = ButtonBaseProps<'button'> & {
```

Comment:
Here we extend ButtonBaseProps types so IconButton can accept all style utility props